packages/core/src/transport-http.ts

```diff
@@ -327,7 +328,9 @@ export class StreamableHttpTransport {
             protocolVersion: protocolHeader || SUPPORTED_MCP_PROTOCOL_VERSION,
             clientInfo: (jsonRpcMessage as JsonRpcReq).params,
           };
-          this.sessions.set(sessionId, { meta: sessionMeta });
+          await Promise.resolve(
```

Comment:
heads up that now this could throw, and the top-level `catch` for this function assumes as a default case that we're dealing with a parse error (see around line 370)